pkg/provider/provider.go

```diff
@@ -142,14 +146,41 @@ var (
 	loaded = false
 )
 
-func buildTestEnvironment() { //nolint:funlen
+// Build image with version based on environment variables if provided, else use a default value
+func buildImageWithVersion() string {
+	tnfPartnerRepo := os.Getenv("TNF_PARTNER_REPO")
+	if tnfPartnerRepo == "" {
+		tnfPartnerRepo = tnfPartnerRepoDef
+	}
+	supportImage := os.Getenv("SUPPORT_IMAGE")
+	if supportImage == "" {
+		supportImage = supportImageDef
+	}
+
+	return tnfPartnerRepo + "/" + supportImage
+}
+
+func buildTestEnvironment() { //nolint:funlen,gocyclo
 	start := time.Now()
+	env = TestEnvironment{}
 	// Wait for the debug pods to be ready before the autodiscovery starts.
-	err := k8sPriviledgedDs.WaitDaemonsetReady(DaemonSetNamespace, DaemonSetName, debugPodsTimeout)
+	oc := clientsholder.GetClientsHolder()
```

Comment:
I'd encapsulate all these statements to deploy the daemon set in a function that returns `bool` in order to fill `DaemonsetFailedToSpawn`. Something like: `env.DaemonsetFailedToSpawn = deployDaemonSet()`.

Reply:
Agreed.  This section was copy/pasted from the older `internal/daemonset/daemonset.go` file.  Can be adjusted in a subsequent PR.